{
    "SchemaVersion": 2,
    "CreatedAt": "2025-04-23T09:25:32.290855242Z",
    "ArtifactName": "registry.satnusa.com/developer/django-helloworld:v1.0.26",
    "ArtifactType": "container_image",
    "Metadata": {
      "OS": {
        "Family": "alpine",
        "Name": "3.21.3"
      },
      "ImageID": "sha256:350dfe862cea85609919c2890038ad6d2fa021cacaca3e2cdd0fe718d0ea36bc",
      "DiffIDs": [
        "sha256:08000c18d16dadf9553d747a58cf44023423a9ab010aab96cf263d2216b8b350",
        "sha256:1ddabb7c6bec326e8e1737665491d708fde945f38ebf4f596589e0637fccce41",
        "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7",
        "sha256:efd2d69eb0955fbcd76bd6735e6640da0335981e268a7ad4445697d6d1513e9e",
        "sha256:d4075cd195d5cd447bf6ae61d378ea3c564fabe033f8e8affd4a01225b8f45a5",
        "sha256:342f082851ea7e3ae514653643610c8b29992ae4b501d640bcba0cd804ba822f",
        "sha256:412fb9775e856d85964522dcd8bbafc01df42af146590f25588e83fc527169a2",
        "sha256:7cee638a499fb36057186827d3d2f4df027e2532c9799a512a848c0531c41408"
      ],
      "RepoTags": [
        "registry.satnusa.com/developer/django-helloworld:v1.0.26"
      ],
      "RepoDigests": [
        "registry.satnusa.com/developer/django-helloworld@sha256:3d623a4a4528df634bdff741863efad6d2f8bf12ad0b8c7b22ddb34fd281f14d"
      ],
      "ImageConfig": {
        "architecture": "amd64",
        "created": "2025-04-23T07:02:30.439932177Z",
        "history": [
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ADD alpine-minirootfs-3.21.3-x86_64.tar.gz / # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "CMD [\"/bin/sh\"]",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ENV PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ENV LANG=C.UTF-8",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "RUN /bin/sh -c set -eux; \tapk add --no-cache \t\tca-certificates \t\ttzdata \t; # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ENV GPG_KEY=E3FF2839C048B25C084DEBE9B26995E310250568",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ENV PYTHON_VERSION=3.9.21",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "ENV PYTHON_SHA256=3126f59592c9b0d798584755f2bf7b081fa1ca35ce7a6fea980108d752a05bb1",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "RUN /bin/sh -c set -eux; \t\tapk add --no-cache --virtual .build-deps \t\tgnupg \t\ttar \t\txz \t\t\t\tbluez-dev \t\tbzip2-dev \t\tdpkg-dev dpkg \t\tfindutils \t\tgcc \t\tgdbm-dev \t\tlibc-dev \t\tlibffi-dev \t\tlibnsl-dev \t\tlibtirpc-dev \t\tlinux-headers \t\tmake \t\tncurses-dev \t\topenssl-dev \t\tpax-utils \t\treadline-dev \t\tsqlite-dev \t\ttcl-dev \t\ttk \t\ttk-dev \t\tutil-linux-dev \t\txz-dev \t\tzlib-dev \t; \t\twget -O python.tar.xz \"https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz\"; \techo \"$PYTHON_SHA256 *python.tar.xz\" | sha256sum -c -; \twget -O python.tar.xz.asc \"https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz.asc\"; \tGNUPGHOME=\"$(mktemp -d)\"; export GNUPGHOME; \tgpg --batch --keyserver hkps://keys.openpgp.org --recv-keys \"$GPG_KEY\"; \tgpg --batch --verify python.tar.xz.asc python.tar.xz; \tgpgconf --kill all; \trm -rf \"$GNUPGHOME\" python.tar.xz.asc; \tmkdir -p /usr/src/python; \ttar --extract --directory /usr/src/python --strip-components=1 --file python.tar.xz; \trm python.tar.xz; \t\tcd /usr/src/python; \tgnuArch=\"$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)\"; \t./configure \t\t--build=\"$gnuArch\" \t\t--enable-loadable-sqlite-extensions \t\t--enable-option-checking=fatal \t\t--enable-shared \t\t--with-ensurepip \t; \tnproc=\"$(nproc)\"; \tEXTRA_CFLAGS=\"-DTHREAD_STACK_SIZE=0x100000\"; \tLDFLAGS=\"${LDFLAGS:--Wl},--strip-all\"; \tmake -j \"$nproc\" \t\t\"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}\" \t\t\"LDFLAGS=${LDFLAGS:-}\" \t; \trm python; \tmake -j \"$nproc\" \t\t\"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}\" \t\t\"LDFLAGS=${LDFLAGS:--Wl},-rpath='\\$\\$ORIGIN/../lib'\" \t\tpython \t; \tmake install; \t\tcd /; \trm -rf /usr/src/python; \t\tfind /usr/local -depth \t\t\\( \t\t\t\\( -type d -a \\( -name test -o -name tests -o -name idle_test \\) \\) \t\t\t-o \\( -type f -a \\( -name '*.pyc' -o -name '*.pyo' -o -name 'libpython*.a' \\) \\) \t\t\\) -exec rm -rf '{}' + \t; \t\tfind /usr/local -type f -executable -not \\( -name '*tkinter*' \\) -exec scanelf --needed --nobanner --format '%n#p' '{}' ';' \t\t| tr ',' '\\n' \t\t| sort -u \t\t| awk 'system(\"[ -e /usr/local/lib/\" $1 \" ]\") == 0 { next } { print \"so:\" $1 }' \t\t| xargs -rt apk add --no-network --virtual .python-rundeps \t; \tapk del --no-network .build-deps; \t\texport PYTHONDONTWRITEBYTECODE=1; \tpython3 --version; \t\tpip3 install \t\t--disable-pip-version-check \t\t--no-cache-dir \t\t--no-compile \t\t'setuptools==58.1.0' \t\twheel \t; \tpip3 --version # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "RUN /bin/sh -c set -eux; \tfor src in idle3 pip3 pydoc3 python3 python3-config; do \t\tdst=\"$(echo \"$src\" | tr -d 3)\"; \t\t[ -s \"/usr/local/bin/$src\" ]; \t\t[ ! -e \"/usr/local/bin/$dst\" ]; \t\tln -svT \"$src\" \"/usr/local/bin/$dst\"; \tdone # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2024-12-05T13:19:52Z",
            "created_by": "CMD [\"python3\"]",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2025-04-23T02:58:05Z",
            "created_by": "WORKDIR /app",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2025-04-23T07:02:00Z",
            "created_by": "COPY . . # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2025-04-23T07:02:27Z",
            "created_by": "RUN /bin/sh -c pip install --no-cache-dir -r requirements.txt # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2025-04-23T07:02:30Z",
            "created_by": "EXPOSE map[80/tcp:{}]",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          },
          {
            "created": "2025-04-23T07:02:30Z",
            "created_by": "RUN /bin/sh -c python manage.py migrate # buildkit",
            "comment": "buildkit.dockerfile.v0"
          },
          {
            "created": "2025-04-23T07:02:30Z",
            "created_by": "CMD [\"python\" \"manage.py\" \"runserver\" \"0.0.0.0:80\"]",
            "comment": "buildkit.dockerfile.v0",
            "empty_layer": true
          }
        ],
        "os": "linux",
        "rootfs": {
          "type": "layers",
          "diff_ids": [
            "sha256:08000c18d16dadf9553d747a58cf44023423a9ab010aab96cf263d2216b8b350",
            "sha256:1ddabb7c6bec326e8e1737665491d708fde945f38ebf4f596589e0637fccce41",
            "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7",
            "sha256:efd2d69eb0955fbcd76bd6735e6640da0335981e268a7ad4445697d6d1513e9e",
            "sha256:d4075cd195d5cd447bf6ae61d378ea3c564fabe033f8e8affd4a01225b8f45a5",
            "sha256:342f082851ea7e3ae514653643610c8b29992ae4b501d640bcba0cd804ba822f",
            "sha256:412fb9775e856d85964522dcd8bbafc01df42af146590f25588e83fc527169a2",
            "sha256:7cee638a499fb36057186827d3d2f4df027e2532c9799a512a848c0531c41408"
          ]
        },
        "config": {
          "Cmd": [
            "python",
            "manage.py",
            "runserver",
            "0.0.0.0:80"
          ],
          "Env": [
            "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "LANG=C.UTF-8",
            "GPG_KEY=E3FF2839C048B25C084DEBE9B26995E310250568",
            "PYTHON_VERSION=3.9.21",
            "PYTHON_SHA256=3126f59592c9b0d798584755f2bf7b081fa1ca35ce7a6fea980108d752a05bb1"
          ],
          "WorkingDir": "/app",
          "ExposedPorts": {
            "80": {}
          },
          "ArgsEscaped": true
        }
      }
    },
    "Results": [
      {
        "Target": "registry.satnusa.com/developer/django-helloworld:v1.0.26 (alpine 3.21.3)",
        "Class": "os-pkgs",
        "Type": "alpine",
        "Vulnerabilities": [
          {
            "VulnerabilityID": "CVE-2025-29087",
            "PkgID": "sqlite-libs@3.48.0-r0",
            "PkgName": "sqlite-libs",
            "PkgIdentifier": {
              "PURL": "pkg:apk/alpine/sqlite-libs@3.48.0-r0?arch=x86_64&distro=3.21.3",
              "UID": "d3184f4c955c09b4"
            },
            "InstalledVersion": "3.48.0-r0",
            "FixedVersion": "3.48.0-r1",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7"
            },
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2025-29087",
            "DataSource": {
              "ID": "alpine",
              "Name": "Alpine Secdb",
              "URL": "https://secdb.alpinelinux.org/"
            },
            "Title": "sqlite: Integer Overflow in SQLite concat_ws Function",
            "Description": "In SQLite 3.44.0 through 3.49.0 before 3.49.1, the concat_ws() SQL function can cause memory to be written beyond the end of a malloc-allocated buffer. If the separator argument is attacker-controlled and has a large string (e.g., 2MB or more), an integer overflow occurs in calculating the size of the result buffer, and thus malloc may not allocate enough memory.",
            "Severity": "MEDIUM",
            "CweIDs": [
              "CWE-190"
            ],
            "VendorSeverity": {
              "bitnami": 1,
              "redhat": 2,
              "ubuntu": 2
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:N/UI:N/S:C/C:N/I:N/A:L",
                "V3Score": 3.2
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 5.5
              }
            },
            "References": [
              "https://access.redhat.com/security/cve/CVE-2025-29087",
              "https://gist.github.com/ylwango613/a44a29f1ef074fa783e29f04a0afd62a",
              "https://nvd.nist.gov/vuln/detail/CVE-2025-29087",
              "https://sqlite.org/releaselog/3_49_1.html",
              "https://www.cve.org/CVERecord?id=CVE-2025-29087",
              "https://www.sqlite.org/cves.html"
            ],
            "PublishedDate": "2025-04-07T20:15:20.253Z",
            "LastModifiedDate": "2025-04-15T16:16:06.233Z"
          },
          {
            "VulnerabilityID": "CVE-2025-31115",
            "PkgID": "xz-libs@5.6.3-r0",
            "PkgName": "xz-libs",
            "PkgIdentifier": {
              "PURL": "pkg:apk/alpine/xz-libs@5.6.3-r0?arch=x86_64&distro=3.21.3",
              "UID": "6a1b930ba614c87c"
            },
            "InstalledVersion": "5.6.3-r0",
            "FixedVersion": "5.6.3-r1",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7"
            },
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2025-31115",
            "DataSource": {
              "ID": "alpine",
              "Name": "Alpine Secdb",
              "URL": "https://secdb.alpinelinux.org/"
            },
            "Title": "xz: XZ has a heap-use-after-free bug in threaded .xz decoder",
            "Description": "XZ Utils provide a general-purpose data-compression library plus command-line tools. In XZ Utils 5.3.3alpha to 5.8.0, the multithreaded .xz decoder in liblzma has a bug where invalid input can at least result in a crash. The effects include heap use after free and writing to an address based on the null pointer plus an offset. Applications and libraries that use the lzma_stream_decoder_mt function are affected. The bug has been fixed in XZ Utils 5.8.1, and the fix has been committed to the v5.4, v5.6, v5.8, and master branches in the xz Git repository. No new release packages will be made from the old stable branches, but a standalone patch is available that applies to all affected releases.",
            "Severity": "HIGH",
            "CweIDs": [
              "CWE-366",
              "CWE-416",
              "CWE-476",
              "CWE-826"
            ],
            "VendorSeverity": {
              "photon": 3,
              "redhat": 3,
              "ubuntu": 2
            },
            "CVSS": {
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 7.5
              }
            },
            "References": [
              "http://www.openwall.com/lists/oss-security/2025/04/03/1",
              "http://www.openwall.com/lists/oss-security/2025/04/03/2",
              "http://www.openwall.com/lists/oss-security/2025/04/03/3",
              "https://access.redhat.com/security/cve/CVE-2025-31115",
              "https://github.com/tukaani-project/xz/commit/d5a2ffe41bb77b918a8c96084885d4dbe4bf6480",
              "https://github.com/tukaani-project/xz/security/advisories/GHSA-6cc8-p5mm-29w2",
              "https://nvd.nist.gov/vuln/detail/CVE-2025-31115",
              "https://tukaani.org/xz/xz-cve-2025-31115.patch",
              "https://ubuntu.com/security/notices/USN-7414-1",
              "https://www.cve.org/CVERecord?id=CVE-2025-31115"
            ],
            "PublishedDate": "2025-04-03T17:15:30.54Z",
            "LastModifiedDate": "2025-04-07T14:18:34.453Z"
          }
        ]
      },
      {
        "Target": "Python",
        "Class": "lang-pkgs",
        "Type": "python-pkg",
        "Vulnerabilities": [
          {
            "VulnerabilityID": "CVE-2023-5752",
            "PkgName": "pip",
            "PkgPath": "app/venv/lib/python3.11/site-packages/pip-23.0.1.dist-info/METADATA",
            "PkgIdentifier": {
              "PURL": "pkg:pypi/pip@23.0.1",
              "UID": "f85c101efa8924ae"
            },
            "InstalledVersion": "23.0.1",
            "FixedVersion": "23.3",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:342f082851ea7e3ae514653643610c8b29992ae4b501d640bcba0cd804ba822f"
            },
            "SeveritySource": "ghsa",
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-5752",
            "DataSource": {
              "ID": "ghsa",
              "Name": "GitHub Security Advisory pip",
              "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
            },
            "Title": "pip: Mercurial configuration injectable in repo revision when installing via pip",
            "Description": "When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.",
            "Severity": "MEDIUM",
            "CweIDs": [
              "CWE-77"
            ],
            "VendorSeverity": {
              "amazon": 2,
              "azure": 1,
              "bitnami": 1,
              "ghsa": 2,
              "nvd": 1,
              "redhat": 1
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              },
              "ghsa": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N",
                "V3Score": 5.5
              },
              "nvd": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              }
            },
            "References": [
              "https://access.redhat.com/security/cve/CVE-2023-5752",
              "https://github.com/pypa/advisory-database/tree/main/vulns/pip/PYSEC-2023-228.yaml",
              "https://github.com/pypa/pip",
              "https://github.com/pypa/pip/commit/389cb799d0da9a840749fcd14878928467ed49b4",
              "https://github.com/pypa/pip/pull/12306",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/",
              "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL",
              "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/",
              "https://nvd.nist.gov/vuln/detail/CVE-2023-5752",
              "https://www.cve.org/CVERecord?id=CVE-2023-5752"
            ],
            "PublishedDate": "2023-10-25T18:17:44.867Z",
            "LastModifiedDate": "2025-02-13T18:16:01.057Z"
          },
          {
            "VulnerabilityID": "CVE-2023-5752",
            "PkgName": "pip",
            "PkgPath": "usr/local/lib/python3.9/site-packages/pip-23.0.1.dist-info/METADATA",
            "PkgIdentifier": {
              "PURL": "pkg:pypi/pip@23.0.1",
              "UID": "db4b2871e1f85d98"
            },
            "InstalledVersion": "23.0.1",
            "FixedVersion": "23.3",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7"
            },
            "SeveritySource": "ghsa",
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-5752",
            "DataSource": {
              "ID": "ghsa",
              "Name": "GitHub Security Advisory pip",
              "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
            },
            "Title": "pip: Mercurial configuration injectable in repo revision when installing via pip",
            "Description": "When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.",
            "Severity": "MEDIUM",
            "CweIDs": [
              "CWE-77"
            ],
            "VendorSeverity": {
              "amazon": 2,
              "azure": 1,
              "bitnami": 1,
              "ghsa": 2,
              "nvd": 1,
              "redhat": 1
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              },
              "ghsa": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N",
                "V3Score": 5.5
              },
              "nvd": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
                "V3Score": 3.3
              }
            },
            "References": [
              "https://access.redhat.com/security/cve/CVE-2023-5752",
              "https://github.com/pypa/advisory-database/tree/main/vulns/pip/PYSEC-2023-228.yaml",
              "https://github.com/pypa/pip",
              "https://github.com/pypa/pip/commit/389cb799d0da9a840749fcd14878928467ed49b4",
              "https://github.com/pypa/pip/pull/12306",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/",
              "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL",
              "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/",
              "https://nvd.nist.gov/vuln/detail/CVE-2023-5752",
              "https://www.cve.org/CVERecord?id=CVE-2023-5752"
            ],
            "PublishedDate": "2023-10-25T18:17:44.867Z",
            "LastModifiedDate": "2025-02-13T18:16:01.057Z"
          },
          {
            "VulnerabilityID": "CVE-2022-40897",
            "PkgName": "setuptools",
            "PkgPath": "usr/local/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/METADATA",
            "PkgIdentifier": {
              "PURL": "pkg:pypi/setuptools@58.1.0",
              "UID": "a31a7b733078920c"
            },
            "InstalledVersion": "58.1.0",
            "FixedVersion": "65.5.1",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7"
            },
            "SeveritySource": "ghsa",
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-40897",
            "DataSource": {
              "ID": "ghsa",
              "Name": "GitHub Security Advisory pip",
              "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
            },
            "Title": "pypa-setuptools: Regular Expression Denial of Service (ReDoS) in package_index.py",
            "Description": "Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.",
            "Severity": "HIGH",
            "CweIDs": [
              "CWE-1333"
            ],
            "VendorSeverity": {
              "alma": 2,
              "amazon": 2,
              "bitnami": 2,
              "cbl-mariner": 2,
              "ghsa": 3,
              "nvd": 2,
              "oracle-oval": 2,
              "photon": 2,
              "redhat": 2,
              "rocky": 2,
              "ubuntu": 2
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 5.9
              },
              "ghsa": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 7.5
              },
              "nvd": {
                "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 5.9
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "V3Score": 5.9
              }
            },
            "References": [
              "https://access.redhat.com/errata/RHSA-2023:0952",
              "https://access.redhat.com/security/cve/CVE-2022-40897",
              "https://bugzilla.redhat.com/2158559",
              "https://bugzilla.redhat.com/show_bug.cgi?id=2158559",
              "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40897",
              "https://errata.almalinux.org/9/ALSA-2023-0952.html",
              "https://errata.rockylinux.org/RLSA-2023:0952",
              "https://github.com/pypa/advisory-database/tree/main/vulns/setuptools/PYSEC-2022-43012.yaml",
              "https://github.com/pypa/setuptools",
              "https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200",
              "https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be",
              "https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1",
              "https://github.com/pypa/setuptools/issues/3659",
              "https://linux.oracle.com/cve/CVE-2022-40897.html",
              "https://linux.oracle.com/errata/ELSA-2024-2987.html",
              "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R",
              "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/",
              "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H",
              "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R",
              "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H",
              "https://nvd.nist.gov/vuln/detail/CVE-2022-40897",
              "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages",
              "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/",
              "https://pyup.io/vulnerabilities/CVE-2022-40897/52495",
              "https://pyup.io/vulnerabilities/CVE-2022-40897/52495/",
              "https://security.netapp.com/advisory/ntap-20230214-0001",
              "https://security.netapp.com/advisory/ntap-20230214-0001/",
              "https://security.netapp.com/advisory/ntap-20240621-0006",
              "https://security.netapp.com/advisory/ntap-20240621-0006/",
              "https://setuptools.pypa.io/en/latest",
              "https://ubuntu.com/security/notices/USN-5817-1",
              "https://www.cve.org/CVERecord?id=CVE-2022-40897"
            ],
            "PublishedDate": "2022-12-23T00:15:13.987Z",
            "LastModifiedDate": "2024-11-21T07:22:13.787Z"
          },
          {
            "VulnerabilityID": "CVE-2024-6345",
            "PkgName": "setuptools",
            "PkgPath": "usr/local/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/METADATA",
            "PkgIdentifier": {
              "PURL": "pkg:pypi/setuptools@58.1.0",
              "UID": "a31a7b733078920c"
            },
            "InstalledVersion": "58.1.0",
            "FixedVersion": "70.0.0",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:4f7cd6c868d53fd0e2fe2fb0df1e1f5f4e3084c4468b7491a54ab459a3a552d7"
            },
            "SeveritySource": "ghsa",
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-6345",
            "DataSource": {
              "ID": "ghsa",
              "Name": "GitHub Security Advisory pip",
              "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
            },
            "Title": "pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools",
            "Description": "A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.",
            "Severity": "HIGH",
            "CweIDs": [
              "CWE-94"
            ],
            "VendorSeverity": {
              "alma": 3,
              "amazon": 3,
              "azure": 3,
              "bitnami": 3,
              "ghsa": 3,
              "oracle-oval": 3,
              "photon": 3,
              "redhat": 3,
              "rocky": 3,
              "ubuntu": 2
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              },
              "ghsa": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              }
            },
            "References": [
              "https://access.redhat.com/errata/RHSA-2024:6726",
              "https://access.redhat.com/security/cve/CVE-2024-6345",
              "https://bugzilla.redhat.com/2297771",
              "https://bugzilla.redhat.com/show_bug.cgi?id=2297771",
              "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-6345",
              "https://errata.almalinux.org/9/ALSA-2024-6726.html",
              "https://errata.rockylinux.org/RLSA-2024:6726",
              "https://github.com/pypa/setuptools",
              "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0",
              "https://github.com/pypa/setuptools/pull/4332",
              "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5",
              "https://linux.oracle.com/cve/CVE-2024-6345.html",
              "https://linux.oracle.com/errata/ELSA-2024-6726.html",
              "https://nvd.nist.gov/vuln/detail/CVE-2024-6345",
              "https://ubuntu.com/security/notices/USN-7002-1",
              "https://www.cve.org/CVERecord?id=CVE-2024-6345"
            ],
            "PublishedDate": "2024-07-15T01:15:01.73Z",
            "LastModifiedDate": "2024-11-21T09:49:28.17Z"
          },
          {
            "VulnerabilityID": "CVE-2024-6345",
            "PkgName": "setuptools",
            "PkgPath": "app/venv/lib/python3.11/site-packages/setuptools-66.1.1.dist-info/METADATA",
            "PkgIdentifier": {
              "PURL": "pkg:pypi/setuptools@66.1.1",
              "UID": "59f8e3ae65e1f805"
            },
            "InstalledVersion": "66.1.1",
            "FixedVersion": "70.0.0",
            "Status": "fixed",
            "Layer": {
              "DiffID": "sha256:342f082851ea7e3ae514653643610c8b29992ae4b501d640bcba0cd804ba822f"
            },
            "SeveritySource": "ghsa",
            "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-6345",
            "DataSource": {
              "ID": "ghsa",
              "Name": "GitHub Security Advisory pip",
              "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
            },
            "Title": "pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools",
            "Description": "A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.",
            "Severity": "HIGH",
            "CweIDs": [
              "CWE-94"
            ],
            "VendorSeverity": {
              "alma": 3,
              "amazon": 3,
              "azure": 3,
              "bitnami": 3,
              "ghsa": 3,
              "oracle-oval": 3,
              "photon": 3,
              "redhat": 3,
              "rocky": 3,
              "ubuntu": 2
            },
            "CVSS": {
              "bitnami": {
                "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              },
              "ghsa": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              },
              "redhat": {
                "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "V3Score": 8.8
              }
            },
            "References": [
              "https://access.redhat.com/errata/RHSA-2024:6726",
              "https://access.redhat.com/security/cve/CVE-2024-6345",
              "https://bugzilla.redhat.com/2297771",
              "https://bugzilla.redhat.com/show_bug.cgi?id=2297771",
              "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-6345",
              "https://errata.almalinux.org/9/ALSA-2024-6726.html",
              "https://errata.rockylinux.org/RLSA-2024:6726",
              "https://github.com/pypa/setuptools",
              "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0",
              "https://github.com/pypa/setuptools/pull/4332",
              "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5",
              "https://linux.oracle.com/cve/CVE-2024-6345.html",
              "https://linux.oracle.com/errata/ELSA-2024-6726.html",
              "https://nvd.nist.gov/vuln/detail/CVE-2024-6345",
              "https://ubuntu.com/security/notices/USN-7002-1",
              "https://www.cve.org/CVERecord?id=CVE-2024-6345"
            ],
            "PublishedDate": "2024-07-15T01:15:01.73Z",
            "LastModifiedDate": "2024-11-21T09:49:28.17Z"
          }
        ]
      }
    ]
  }
